bpython/autocomplete.py

```diff
@@ -335,6 +335,48 @@ def list_attributes(self, obj):
                 return dir(obj)
 
 
+class ArrayItemMembersCompletion(BaseCompletionType):
+
+    def __init__(self, shown_before_tab=True, mode=SIMPLE):
+        self._shown_before_tab = shown_before_tab
+        self.completers = CumulativeCompleter((
+                            StringLiteralAttrCompletion(mode=mode),
+                            AttrCompletion(mode=mode),
+                            ParameterNameCompletion(mode=mode)),
+                            mode=mode)
+
+    def matches(self, cursor_offset, line, **kwargs):
+        if 'locals_' not in kwargs:
+            return None
+        locals_ = kwargs['locals_']
+
+        r = self.locate(cursor_offset, line)
+        if r is None:
+            return None
+        member_part = r[2]
+        _, _, dexpr = lineparts.current_array_with_indexer(cursor_offset, line)
+        try:
+            exec('temp_val_from_array = ' + dexpr, locals_)
```

Comment:
I went ahead and submitted some updates. I got it using safe_eval if you're fine with that approach. I just had to set locals_ acting as a dictionary, equal to the safe_eval.